src/parser/transform/transform_expression.cpp

```diff
@@ -438,8 +439,14 @@ std::unique_ptr<ParsedExpression> Transformer::transformListLiteral(
 
 std::unique_ptr<ParsedExpression> Transformer::transformStructLiteral(
     CypherParser::KU_StructLiteralContext& ctx) {
+    auto numFields = ctx.kU_StructField().size();
```

Comment:
Instead of adding the check for number of fields there, I would implement the check inside the binder.

Reply:
Yes just move the logic into LogicalType::convertFromString